app/controllers/phylo_trees_controller.rb

```diff
@@ -120,27 +137,41 @@ def set_phylo_tree
     assert_access
   end
 
-  def sample_details_json(pipeline_runs, taxid)
+  def sample_details_json(pipeline_run_ids, taxid)
+    return [] if pipeline_run_ids.blank?
+
     # Retrieve information for displaying the tree's sample list.
     # Expose it as an array of hashes containing
     # - sample name
     # - project id and name
     # - pipeline run id to be used for the sample
-    # - number of reads matching the specified taxid in NT
-    return [] if pipeline_runs.blank?
-    Sample.connection.select_all("
-                   select pipeline_runs_samples_projects.name,
-                          pipeline_runs_samples_projects.project_id,
-                          pipeline_runs_samples_projects.project_name,
-                          pipeline_runs_samples_projects.pipeline_run_id,
-                          taxon_counts.count as taxid_nt_reads
-                   from (select pipeline_runs.id as pipeline_run_id, samples.name, samples.project_id, projects.name as project_name
-                         from ((pipeline_runs inner join samples on pipeline_runs.sample_id = samples.id)
-                               inner join projects on samples.project_id = projects.id)
-                         where pipeline_runs.id in (#{pipeline_runs.pluck(:id).join(',')})) pipeline_runs_samples_projects
-                   inner join taxon_counts
-                   on taxon_counts.pipeline_run_id = pipeline_runs_samples_projects.pipeline_run_id
-                   where taxon_counts.tax_id = #{taxid} and taxon_counts.count_type = 'NT'
+    samples_projects = Sample.connection.select_all("
+      select
+        pipeline_runs_samples_projects.name,
+        pipeline_runs_samples_projects.project_id,
+        pipeline_runs_samples_projects.project_name,
+        pipeline_runs_samples_projects.pipeline_run_id
+      from
+        (select
+           pipeline_runs.id as pipeline_run_id,
+           samples.name,
+           samples.project_id,
+           projects.name as project_name
+         from
+           ((pipeline_runs inner join samples on pipeline_runs.sample_id = samples.id)
+            inner join projects on samples.project_id = projects.id)
+         where pipeline_runs.id in (#{pipeline_run_ids.join(',')})) pipeline_runs_samples_projects
     ").to_hash
+
+    # Also add:
+    # - number of reads matching the specified taxid in NT
+    # Do not include the query on taxon_counts in the previous query above using a join,
+    # because the taxon_counts table is large.
+    taxon_counts = TaxonCount.where(pipeline_run_id: pipeline_run_ids).where(tax_id: taxid).where(count_type: 'NT').index_by(&:pipeline_run_id)
```

Comment:
make sure there's index for querying tax_id in taxon_counts.

Reply:
yes, there is 
```
t.index %w[pipeline_run_id tax_id count_type], name: "taxon_counts_pr_index", unique: true
```

Reply:
https://github.com/chanzuckerberg/idseq-web/blob/master/db/schema.rb#L349